pinot-core/src/test/java/com/linkedin/pinot/query/aggregation/AggregationGroupByWithDictionaryAndTrieTreeOperatorMultiValueTest.java

```diff
@@ -354,69 +353,62 @@ public void testEmptyQueryResultsForInterSegmentAggregationGroupBy() throws Exce
     final DataTable instanceResponse = globalPlan.getInstanceResponse();
     LOGGER.debug("instanceResponse: {}", instanceResponse);
 
-    final DefaultReduceService defaultReduceService = new DefaultReduceService();
+    final BrokerReduceService reduceService = new BrokerReduceService();
     final Map<ServerInstance, DataTable> instanceResponseMap = new HashMap<ServerInstance, DataTable>();
     instanceResponseMap.put(new ServerInstance("localhost:0000"), instanceResponse);
-    final BrokerResponseJSON brokerResponse = defaultReduceService.reduceOnDataTable(brokerRequest, instanceResponseMap);
-    LOGGER.debug("brokerResponse: {}", new JSONArray(brokerResponse.getAggregationResults()));
-    LOGGER.debug("Time used : {}", brokerResponse.getTimeUsedMs());
+    final BrokerResponseNative brokerResponse = reduceService.reduceOnDataTable(brokerRequest, instanceResponseMap);
     assertEmptyBrokerResponse(brokerResponse);
   }
 
-  private void assertBrokerResponse(int numSegments, BrokerResponseJSON brokerResponse) throws JSONException {
+  private void assertBrokerResponse(int numSegments, BrokerResponseNative brokerResponse) {
     Assert.assertEquals(100000 * numSegments, brokerResponse.getNumDocsScanned());
-    final int groupLength = 15;
-
-    verifyResponse(brokerResponse, groupLength);
+    final int groupSize = 15;
+    verifyResponse(brokerResponse, groupSize);
 
     // Assertion on Aggregation Results
-    final List<double[]> aggregationResult = getAggregationResult(numSegments);
+    final List<double[]> aggregationResults = getAggregationResult(numSegments);
     final List<String[]> groupByResult = getGroupResult();
     for (int j = 0; j < _numAggregations; ++j) {
-      LOGGER.debug("For aggregation function: {}", _aggregationInfos.get(j));
-      final double[] aggResult = aggregationResult.get(j);
-      final String[] groupResult = groupByResult.get(j);
+      final double[] aggResult = aggregationResults.get(j);
+      final String[] expectedGroupByResult = groupByResult.get(j);
       for (int i = 0; i < 15; ++i) {
-        LOGGER.debug("Comparing group: {}", i);
+        AggregationResult aggregationResult = brokerResponse.getAggregationResults().get(j);
+        GroupByResult actualGroupByResult = aggregationResult.getGroupByResult().get(i);
         Assert.assertEquals(0, DoubleComparisonUtil.defaultDoubleCompare(aggResult[i],
```

Comment:
Is this equivalent to "Assert.assertEquals(Double.parseDouble(actualGroupByResult.getValue().toString()), aggResult[i])"?